internal/kustomize/kustomize.go

```diff
@@ -80,3 +75,26 @@ func Render(
 	// Glue the manifests together
 	return manifests.CombineYAML(yamlManifests), nil
 }
+
+func BuildKustomizeOptions(
+	enableHelm bool,
+	loadRestrictor string,
+) *argoappv1.KustomizeOptions {
+	buildOptions := ""
+
+	if enableHelm {
+		buildOptions += "--enable-helm"
+	}
+
+	if loadRestrictor != "" {
+		if buildOptions != "" {
+			buildOptions += " "
+		}
+
+		buildOptions += fmt.Sprintf("--load-restrictor %s", loadRestrictor)
+	}
+
+	return &argoappv1.KustomizeOptions{
+		BuildOptions: buildOptions,
+	}
```

Comment:
We could always add the space so we wouldn't need to check whether `buildOptions` is empty:

```
buildOptions += fmt.Sprintf(" --load-restrictor %s", loadRestrictor)
..
return &argoappv1.KustomizeOptions{
    BuildOptions: strings.TrimSpace(buildOptions),
}
```